src/main/java/com/google/sps/utils/User.java

```diff
@@ -0,0 +1,102 @@
+package com.google.sps.utils;
+import com.google.appengine.api.datastore.Entity;
+import java.lang.IllegalArgumentException;
+import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+  
+/**
+ * Wrapper around a datastore Entity representing a user.
+ *
+ * This adds some type safety and relieves the need for type casting when
+ * handling user entities.
+ */
+public final class User {
+
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
It'd be better not to have this class automatically update datastore with each setter call. A few reasons:

* It'll be easier to unit test if it doesn't use datastore.
* Often you'll want datastore updates to be all-or-nothing. For example, imagine you want to add the user to a new game and set their score to zero. If you use two separate calls to datastore, something could go wrong in between, and then they'd be in the new game with a nonzero score.
* If you change several fields at once, it'll be much slower if you're sending a separate request to datastore for each field.

Instead, I recommend using the pattern `datastore.put(user.getEntity())` in the code that uses `User`.

Reply:
That makes sense, I took datastore out.